### PR TITLE
fix/secp256k1

### PR DESCRIPTION
**Pull request by mistermoe**
* Account for signatures less than 64 bytes
* add test
* bump version

### CODE REVIEW

File: crypto/src/main/kotlin/web5/sdk/crypto/Secp256k1.kt [hunk 1]
```diff
@@ -349,4 +344,34 @@ public object Secp256k1 : KeyGenerator, Signer {
...
+ * If the original byte array is shorter than the expected length, the function will pad the array with leading 0 bytes
+ * to reach the required size.
+ *
+ * This adjustment is particularly useful when dealing with cryptographic operations where a fixed size byte array
+ * is expected, ensuring consistency and correctness of the data format.
+ *
+ * @param size The expected fixed length of the resulting byte array.
+ * @return A [ByteArray] of size [size], representing the [BigInteger] value.
+ */
+private fun BigInteger.toFixedByteArray(size: Int): ByteArray {
```
Comment (KendallWeihe) [inline crypto/src/main/kotlin/web5/sdk/crypto/Secp256k1.kt:368]: TIL you can `private` a function outside of a class in Kotlin

File: crypto/src/test/kotlin/web5/sdk/crypto/Secp256k1Test.kt [hunk 2]
```diff
@@ -53,6 +55,34 @@ class Secp256k1Test {
     val base64UrlEncodedSig2 = Convert(sig2).toBase64Url(padding = false)
 
     assertEquals(base64UrlEncodedSig1, base64UrlEncodedSig2)
+  }
+
+  @Test
+  fun fuzz() {
```
Comment (KendallWeihe) [inline crypto/src/test/kotlin/web5/sdk/crypto/Secp256k1Test.kt:61]:
```suggestion
  fun `can verify 10k random payloads`() {
```

or whatevs, not sure what the intent here is

Comment (mistermoe) [reply crypto/src/test/kotlin/web5/sdk/crypto/Secp256k1Test.kt:61]: point here is to generate a sufficient number of signatures (aka 2 big ints) across a large enough surface area to suss out any edge cases there may be with signature verification

**Review by phoebe-lew — APPROVED** [review]

File: crypto/src/test/kotlin/web5/sdk/crypto/Secp256k1Test.kt (hunk 2 above)
Comment (mistermoe) [reply crypto/src/test/kotlin/web5/sdk/crypto/Secp256k1Test.kt:61]: cool with "pressure test signature verification" as the test name?

Comment (KendallWeihe) [reply crypto/src/test/kotlin/web5/sdk/crypto/Secp256k1Test.kt:61]: yup 👍

Comment (mistermoe) [reply crypto/src/test/kotlin/web5/sdk/crypto/Secp256k1Test.kt]: done - https://github.com/TBD54566975/web5-kt/commit/8049d78d49280d274a377d215016c9cff21d56f8

File: crypto/src/test/kotlin/web5/sdk/crypto/Secp256k1Test.kt [hunk 3]
```diff
@@ -53,6 +55,34 @@ class Secp256k1Test {
...
+    repeat(10_000) {
+      val payload = generateRandomString(Random().nextInt(100))
 
+      try {
+        val sig1 = Secp256k1.sign(privateKey, payload.toByteArray())
+        Secp256k1.verify(publicKey, payload.toByteArray(), sig1)
+      } catch (e: SignatureException) {
+        println("($it) $e. Payload: $payload")
+        throw e
+      }
```
Comment (andresuribe87) [inline crypto/src/test/kotlin/web5/sdk/crypto/Secp256k1Test.kt:85]: Perhaps this would be simpler? 
```suggestion
    repeat(10_000) {
      val payload = Random.nextBytes(Random.nextInt(100))

      try {
        val sig1 = Secp256k1.sign(privateKey, payload)
        Secp256k1.verify(publicKey, payload, sig1)
      } catch (e: SignatureException) {
        val payloadString = Convert(payload).toBase64Url(false)
        println("($it) $e. Payload (base64url encoded): $payloadString")
        throw e
      }
```

**Review by andresuribe87 — APPROVED** [review]
lgtm, smol suggestion

Would have been awesome to identify a test that fails before, but passes now.

**Comment by mistermoe** [comment]
> lgtm, smol suggestion
> 
> Would have been awesome to identify a test that fails before, but passes now.

good point @andresuribe87 . will keep this in mind for future PRs! less valuable now but for context:

we were running into intermittent unit test failures on the tbdex side w.r.t. signature verification e.g. 

```
OfferingTest > can parse offering from a json string() FAILED
    java.security.SignatureException: Invalid Signature
        at web5.sdk.crypto.Secp256k1.verify(Secp256k1.kt:280)
        at web5.sdk.crypto.Signer$DefaultImpls.verify$default(Signer.kt:115)
        at web5.sdk.crypto.Crypto.verify(Crypto.kt:144)
```

@tomdaffurn and i narrowed it down to secp256k1 and wrote the test that we added to this PR to consistently reproduce the issue. we used the failing test to track down and fix the bug. 

is this the type of context you were suggesting might be helpful?

File: crypto/src/test/kotlin/web5/sdk/crypto/Secp256k1Test.kt (hunk 3 above)
Comment (mistermoe) [reply crypto/src/test/kotlin/web5/sdk/crypto/Secp256k1Test.kt:85]: awesome suggestion.

**Comment by andresuribe87** [comment]
@mistermoe I meant more to include a test in this PR that always failed before your fix, but passes after the fix. Sounds like you kinda had it? No big deal. 

Context was super useful, thx!
